src/resources/types/surface.c

```diff
@@ -391,6 +394,7 @@ wlc_surface(struct wlc_surface *surface)
        !chck_iter_pool(&surface->subsurface_list, 4, 0, sizeof(wlc_resource)))
       goto fail;
 
+   surface->commit.scale = 1;
```

Comment:
Fair enough. It sounds like 0 scale for initial doesn't make much sense anyways. We consider and check client set surface scale at <= 0 illegal anyways.